npm/ng-packs/packages/theme-shared/src/lib/directives/ngx-datatable-list.directive.ts

```diff
@@ -7,30 +7,40 @@ import {
   DoCheck,
   SimpleChanges,
   inject,
-  DestroyRef
+  DestroyRef,
+  ViewContainerRef,
+  Renderer2,
+  AfterViewInit,
 } from '@angular/core';
 import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
 import { DatatableComponent } from '@swimlane/ngx-datatable';
-import { ListService, LocalizationService } from '@abp/ng.core';
+import { combineLatest } from 'rxjs';
+import { ListService, LocalizationService, RouterWaitService } from '@abp/ng.core';
 import {
   defaultNgxDatatableMessages,
   NGX_DATATABLE_MESSAGES,
 } from '../tokens/ngx-datatable-messages.token';
+import { SpinnerComponent } from '../components';
 
 @Directive({
   // eslint-disable-next-line @angular-eslint/directive-selector
   selector: 'ngx-datatable[list]',
   standalone: true,
   exportAs: 'ngxDatatableList',
 })
-export class NgxDatatableListDirective implements OnChanges, OnInit, DoCheck {
-  @Input() list!: ListService;
-
+export class NgxDatatableListDirective implements OnChanges, OnInit, DoCheck, AfterViewInit {
   protected readonly table = inject(DatatableComponent);
   protected readonly cdRef = inject(ChangeDetectorRef);
   protected readonly destroyRef = inject(DestroyRef);
   protected readonly localizationService = inject(LocalizationService);
   protected readonly ngxDatatableMessages = inject(NGX_DATATABLE_MESSAGES, { optional: true });
+  protected readonly routerWaitService = inject(RouterWaitService);
+  protected readonly viewContainerRef = inject(ViewContainerRef);
+  protected readonly renderer = inject(Renderer2);
+
+  protected _loading = true;
```

Comment:
We've list service already, use inside loading prop of it instead using local var.

Reply:
I have also removed the extra check that I add for `RouterWaitService`